src/main/java/hudson/remoting/PipeWindow.java

```diff
@@ -95,19 +104,23 @@
 
     /**
      * Indicates that the remote end has died and all the further send attempt should fail.
+     * @param cause Death cause. If {@code null}, the death will be still considered as dead, but there will be no cause recorded..
      */
-    void dead(Throwable cause) {
-        this.dead = cause;
+    void dead(@CheckForNull Throwable cause) {
+        // We need to record
+        this.dead = cause != null ? cause : new RemotingSystemException("Unknown cause", null) ;
```

Comment:
Without it the handling was just wrong in the case of the missing cause. It may be a root cause of some defects reported to remoting